File: problem/Triplet_Subsequence/increasingTriplet.cpp
class Solution
{
public:
    bool increasingTriplet(vector<int> &nums)
    {
        int i = INT_MAX, j = INT_MAX;
        for (int k = 0; k < nums.size(); k++)
        {
            if (nums[k] <= i)
                i = nums[k];
            else if (nums[k] <= j)
                j = nums[k];
            else
                return true;
        }
        return false;
    }
};

File: problem/Triplet_Subsequence/increasingTriplet.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var increasingTriplet = function(nums) {
    let i = Infinity, j = Infinity;
    for (let num of nums) {
        if (num <= i) i = num;
        else if (num <= j) j = num;
        else return true;
    }
    return false;
};
